apps/api/test/04-report-4.e2e-spec.ts

```diff
@@ -0,0 +1,160 @@
+import { INestApplication } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import request from 'supertest';
+
+import { AppModule } from 'src/app.module';
+import { AuthGuard } from 'src/auth/auth.guard';
+import { URLS } from './constants';
+import { canActivate } from './override-guard';
+import { getApplicant, getStatus } from './report-util';
+import { LICENSE_STATUS, BCCNM_STATUS, jobInput, Report4Item } from './fixture/reports';
+import { STATUS } from '@ien/common';
+import {
+  addApplicant,
+  addMilestone,
+  setApp,
+  addJob,
+  deleteApplicantStatus,
+} from './report-request-util';
+
+describe('Report 4 - Number of IEN registrants in the licensing stage', () => {
+  // Tracks applicants id's and to later delete status/milestones
+  const applicantIds: string[] = [];
+  let app: INestApplication;
+
+  // Find the number of applicants for the given status
+  const report4NumberOfApplicants = (body: Report4Item[], applicantStatus: string | STATUS) => {
+    return body.find((e: { status: string }) => {
+      return e.status === applicantStatus;
+    })?.applicants;
+  };
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    })
+      .overrideGuard(AuthGuard)
+      .useValue({ canActivate })
+      .compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+
+    setApp(app);
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  it('Status 200 on report 4 response', async () => {
+    const { status } = await request(app.getHttpServer()).get(URLS.REPORT4);
+    expect(status).toBe(200);
+  });
+
+  it('Add applicants and give each a different status/milestone', async () => {
+    const statusIds: string[] = [];
+
+    const { body: before } = await request(app.getHttpServer()).get(URLS.REPORT4);
+
+    // What the report is expected to look like. Used for comparison
+    const expectedResult = before.map((item: { applicants: string; status: string }) => {
+      let result = parseInt(item.applicants) + 1;
+      // Accounts for other two BCCNM Licenses
+      if (
+        item.status === 'Granted provisional licensure' ||
```

Comment:
How about using `STATUS` enum?

Reply:
The hardcoded licensure strings are used here because they don't exist on STATUS. For the withdrawn status, you are correct. I will fix that now.